arch/arc/core/prep_c.c

```diff
@@ -67,6 +69,20 @@ static void invalidate_dcache(void)
 }
 #endif
 
+#ifdef __CCAC__
+/**
+ * @brief Clear device_states section
+ *
+ * This routine clears the device_states section,
+ * as MW compiler marks the section with NOLOAD flag.
+ */
+__boot_func
+static void dev_state_zero(void)
+{
+	(void) memset(__device_states_start, 0, __device_states_end - __device_states_start);
```

Comment:
Hmm, shouldn't we use `z_early_memset` and not just regular lib-c `memset`?

All other init functions (like `z_bss_zero` uses `z_early_memset`).

Reply:
z_early_memset is a wrapper (with correspond modifier) for regular memset:  https://github.com/kokas-a/zephyr/blob/cf053d07e37e9ee59c444af3c635317e17123a76/kernel/init.c#L142. So I suppose there is no need to call wrapper twice